Modules/Devices/test/Devices.Application.Tests/Tests/Identities/Commands/StartDeletionProcessAsSupport/HandlerTests.cs

```diff
@@ -0,0 +1,101 @@
+﻿using Backbone.BuildingBlocks.Application.Abstractions.Exceptions;
+using Backbone.BuildingBlocks.Application.Abstractions.Infrastructure.EventBus;
+using Backbone.Modules.Devices.Application.Identities.Commands.StartDeletionProcessAsSupport;
+using Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Devices.Application.IntegrationEvents.Outgoing;
+using Backbone.Modules.Devices.Domain.Entities.Identities;
+using Backbone.UnitTestTools.Extensions;
+using FakeItEasy;
+using FluentAssertions;
+using Xunit;
+using static Backbone.UnitTestTools.Data.TestDataGenerator;
+
+namespace Backbone.Modules.Devices.Application.Tests.Tests.Identities.Commands.StartDeletionProcessAsSupport;
+
+public class HandlerTests
+{
+    [Fact]
+    public async Task Happy_path_as_support()
+    {
+        // Arrange
+        var activeIdentity = TestDataGenerator.CreateIdentityWithOneDevice();
+
+        var mockIdentitiesRepository = A.Fake<IIdentitiesRepository>();
+        var fakeEventBus = A.Fake<IEventBus>();
+
+        A.CallTo(() => mockIdentitiesRepository.FindByAddress(activeIdentity.Address, A<CancellationToken>._, A<bool>._))
+            .Returns(activeIdentity);
+
+        var handler = CreateHandler(mockIdentitiesRepository, fakeEventBus);
+
+        // Act
+        var response = await handler.Handle(new StartDeletionProcessAsSupportCommand(activeIdentity.Address), CancellationToken.None);
+
+        // Assert
+        response.Should().NotBeNull();
+        response.Status.Should().Be(DeletionProcessStatus.WaitingForApproval);
+
+        A.CallTo(() => mockIdentitiesRepository.Update(
+                A<Identity>.That.Matches(
+                    i => i.Address == activeIdentity.Address &&
+                         i.DeletionProcesses.Count == 1 &&
+                         i.DeletionProcesses[0].Id == response.Id &&
+                         i.DeletionProcesses[0].AuditLog.Count == 1),
+                A<CancellationToken>._))
+            .MustHaveHappenedOnceExactly();
+    }
+
+    [Fact]
+    public async void Publishes_IdentityDeletionProcessStartedEvent()
+    {
+        // Arrange
+        var activeIdentity = TestDataGenerator.CreateIdentityWithOneDevice();
+
+        var mockIdentitiesRepository = A.Fake<IIdentitiesRepository>();
```

Comment:
I don't know how often I will have to say this, but: be careful with the names. This is not a mock.

Reply:
I was basing off the tests for the deletion process as a user, which uses these exact naming conventions.
Since it was already review and approved, I figured I could skip the naming checks.

Reply:
On the handler tests for the deletion process as a user, more specifically in the `Cannot_start_when_given_identity_does_not_exist`, are the naming conventions not wrong?